beacon/goclient/proposer.go

```diff
@@ -166,7 +170,14 @@ func (gc *GoClient) SubmitBlindedBeaconBlock(block *api.VersionedBlindedProposal
 		Proposal: signedBlock,
 	}
 
-	return gc.client.SubmitBlindedProposal(gc.ctx, opts)
+	start := time.Now()
+	if err := gc.client.SubmitBlindedProposal(gc.ctx, opts); err != nil {
+		return err
+	}
+
+	recordRequestDuration(gc.ctx, "SubmitBlindedProposal", gc.client.Address(), http.MethodPost, time.Since(start))
```

Comment:
This comment works a lot of different API call metrics. What if the API call err'ed , some times its useful to know its err'ed and the duration it took the client to return error.

Reply:
The potential issue with recording this metric when an error is returned is that the `SubmitBlindedProposal` method has additional responsibilities beyond the HTTP call, such as input validation, which happens before the HTTP call and may result in an error. The purpose of this histogram is to record _response times for HTTP calls_.
This is a trade-off: by not recording errors that may result from simple input validation, we avoid diluting the histogram. However, this means missing out on network errors and unsuccessful status codes (since the library wraps everything that isn’t a 2XX status code into an error). Of course we can decide to record durations always, even if there is an error returned, that's just the pattern followed across entire codebase. Valid point though.

Reply:
well, what you mentioned also works for successful calls. if you measuring `SubmitBlindedProposal` as a whole you always adding the input checks time inside. maybe we need to measure in lower level if we want only http call for real. Also wouldn't it be easy to not corrupt the histogram data by adding an error label so in dashboard you can look only on successful calls? I guess we could also add reason label and only look at errors that can back from the API as well.

I think this is a direction we want to go in, of course you can draw action items out of it and save it for next versions. @zktaiga @oleg-ssvlabs @moshe-blox .